packages/cadl-ranch-specs/http/azure/core/mockapi.ts

```diff
@@ -37,24 +35,54 @@ Scenarios.Azure_Core_get = passOnSuccess(
       throw new ValidationError("Expected path param id=1", "1", req.params.id);
     }
     req.expect.containsQueryParam("api-version", "2022-12-01-preview");
-    const responseBody = { id: 1, name: "Madge" };
-    return { status: 200, body: json(responseBody) };
+    return { status: 200, body: json(validUser) };
   }),
 );
 
 Scenarios.Azure_Core_list = passOnSuccess(
   mockapi.get("/azure/core/users", (req) => {
     req.expect.containsQueryParam("api-version", "2022-12-01-preview");
+    req.expect.containsQueryParam("top", "5");
+    req.expect.containsQueryParam("skip", "10");
+    req.expect.containsQueryParam("orderby", "id");
+    req.expect.containsQueryParam("filter", "id lt 10");
+    if (!req.originalRequest.originalUrl.includes("select=id&select=orders&select=ETag")) {
+      throw new ValidationError(
+        "Expected query param select=id&select=orders&select=ETag ",
+        "select=id&select=orders&select=ETag",
+        req.headers["select"],
```

Comment:
req.query?

Reply:
checked `req.query["select"]` is an array like ['id', 'orders', 'ETag'].
So updated to use `req.originalRequest.originalUrl`